client/src/app/site/agenda/models/view-speaker.ts

```diff
@@ -0,0 +1,86 @@
+import { BaseViewModel } from 'app/site/base/base-view-model';
+import { Speaker } from 'app/shared/models/agenda/speaker';
+import { User } from 'app/shared/models/users/user';
+import { Selectable } from 'app/shared/components/selectable';
+
+/**
+ * Determine the state of the speaker
+ */
+export enum SpeakerState {
+    WAITING,
+    CURRENT,
+    FINISHED
+}
+
+/**
+ * Provides "safe" access to a speaker with all it's components
+ */
+export class ViewSpeaker extends BaseViewModel implements Selectable {
+    private _speaker: Speaker;
+    private _user: User;
+
+    public get speaker(): Speaker {
+        return this._speaker;
+    }
+
+    public get user(): User {
+        return this._user;
+    }
+
+    public get id(): number {
+        return this.speaker ? this.speaker.id : null;
+    }
+
+    public get weight(): number {
+        return this.speaker ? this.speaker.weight : null;
+    }
+
+    public get marked(): boolean {
+        return this.speaker ? this.speaker.marked : null;
+    }
+
+    public get begin_time(): string {
+        return this.speaker ? this.speaker.begin_time : null;
+    }
+
+    public get end_time(): string {
+        return this.speaker ? this.speaker.end_time : null;
+    }
+
+    public get state(): SpeakerState {
+        if (!this.begin_time && !this.end_time) {
+            return SpeakerState.WAITING;
+        } else if (this.begin_time && !this.end_time) {
+            return SpeakerState.CURRENT;
+        } else {
+            return SpeakerState.FINISHED;
+        }
+    }
+
+    /**
+     * Using short name for now. Could be full name.
```

Comment:
proper `@returns` is missing in documentation

Reply:
Comment should be removed since it's wrong (do you even read) ;-)